server/services/store/sqlstore/migrations/migrations_files/000009_blocks_history.up.sql

```diff
@@ -0,0 +1,30 @@
+ALTER TABLE {{.prefix}}blocks RENAME TO {{.prefix}}blocks_history;
+CREATE TABLE IF NOT EXISTS {{.prefix}}blocks (
+	id VARCHAR(36),
+	{{if .postgres}}insert_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),{{end}}
+	{{if .sqlite}}insert_at DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),{{end}}
+	{{if .mysql}}insert_at DATETIME(6) NOT NULL DEFAULT NOW(6),{{end}}
+	parent_id VARCHAR(36),
+	workspace_id VARCHAR(36),
+	root_id VARCHAR(36),
+	modified_by VARCHAR(36),
+	`schema` BIGINT,
+	type TEXT,
+	title TEXT,
+	fields {{if .postgres}}JSON{{else}}TEXT{{end}},
+	create_at BIGINT,
+	update_at BIGINT,
+	delete_at BIGINT,
+	PRIMARY KEY (id)
+){{if .mysql}}CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci{{end}};
+
+{{if .mysql}}
+INSERT IGNORE INTO {{.prefix}}blocks (SELECT * FROM {{.prefix}}blocks_history ORDER BY insert_at DESC);
```

Comment:
Not sure if this is actionable here, but it looks like the .prefix setting should be constant for a given instance, otherwise migrations would break in unexpected ways. Maybe more a note for the future that .prefix should ideally be a hardcoded build setting, or something similar.

Reply:
maybe yes, maybe not, the behavior if you change the prefix, because the migrations table is prefixed to, is treat that situation as a brand new database, the prefix doesn't change on runtime, only changes on restart, so... shouldn't be a big deal.